### PR TITLE
Cleanup unused predicate error types.

### LINKED ISSUE #86988: Cleanup unused predicate error types [Migration Phase 2]
The following predicates error types are deprecated and should be removed:

ErrNodeNetworkUnavailable
ErrNodeUnderMemoryPressure
ErrNodeUnderDiskPressure
ErrNodeUnderPIDPressure
ErrFakePredicate
ErrNodeNetworkUnavailable
ErrNodeNotReady

Part of #85822

/sig scheduling
/help

### PR DESCRIPTION
**Pull request by denkensk**
**What type of PR is this?**
/kind cleanup
/sig scheduling

**What this PR does / why we need it**:
Cleanup unused predicate error types.

**Which issue(s) this PR fixes**:
Fixes #86988 

**Does this PR introduce a user-facing change?**:
```release-note
None
```

### CODE REVIEW

File: pkg/scheduler/core/generic_scheduler_test.go [hunk 1]
```diff
@@ -66,6 +66,8 @@ var (
 	errPrioritize = fmt.Errorf("priority map encounters an error")
 )
 
+const ErrFakePredicate = "Nodes failed the fake predicate"
```
Comment (ahg-g) [inline pkg/scheduler/core/generic_scheduler_test.go]: Name it ErrReasonFake, predicates are deprecated

**Review by ahg-g — COMMENTED** [review]
One nit

File: pkg/scheduler/core/generic_scheduler_test.go (hunk 1 above)
Comment (denkensk) [reply pkg/scheduler/core/generic_scheduler_test.go]: Done  thx~

**Pull request merged**